src/VisualStudio/Core/Test.Next/Mocks/InProcRemoteHostClientFactory.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Remote;
+using Roslyn.Test.Utilities.Remote;
+
+namespace Roslyn.VisualStudio.Next.UnitTests.Mocks
+{
+    [ExportWorkspaceService(typeof(IRemoteHostClientFactory), layer: ServiceLayer.Host), Shared]
+    internal class InProcRemoteHostClientFactory : IRemoteHostClientFactory
+    {
+        public Task<RemoteHostClient> CreateAsync(Workspace workspace, CancellationToken cancellationToken)
+        {
+            return InProcRemoteHostClient.CreateAsync(workspace, cancellationToken);
```

Comment:
Does this create a fresh client each time?  Do we isntead want a single client htat can be shared among all tests?  That would help us test how the client works under a heavy multi-threaded workload.